wordpress-mobile/WordPress-iOS
# Fix Position of Search Icon in Site Creation - Choose a Domain for RTL languages

# 📱 Fix Position of Search Icon in RTL languages

Fixes [#13607 Search icon appears on left side in RTL language](https://github.com/wordpress-mobile/WordPress-iOS/issues/13607)

This pull-request fixes the input field in **Site Creation &rarr; Choose a Domain** for RTL languages:
- Position of search / loading icon
- Position of clear button
- Text alignment of the editable part in the text field

I took inspiration from the default iOS layout for search fields in RTL languages:

![expected_icon_rtl](https://user-images.githubusercontent.com/4588074/155154058-d2720c0b-ad84-4839-9d4f-cb8e99bb4d3b.png)

## To Test

1. Make sure the language of the device is set to a right to left (RTL) language (Arabic, Hebrew, etc)
If building the app from this pull request in Xcode, you can go to Product → Scheme → Edit Scheme and set the app language to Arabic or Hebrew before building.
2. Start creating a new site using the app (From My Site screen tap the ▼ icon next to the site title)
3. On the Site Picker screen tap on the ➕ menu button
4. Select **Create WordPress.com site** (first option)
5. Choose any design or tap the **SKIP** menu button (doesn't matter for this test)
6. **Verify**
   1. Search icon is positioned at the right of the text
   2. Input placeholder text is aligned to the right
7. Type  anything random for the domain, like `example test`
8. **Verify** 
   1. Loading icon appears at the right of the text (where the search icon was)
   2. After loading results, the search icon is positioned at the right
   3. The text that you entered is aligned to the right
   4. The clear button is positioned at the left of the text input

## Previews

| Before | After |
| ------ | ------ |
| ![before](https://user-images.githubusercontent.com/4588074/155291857-f0b4cde2-c5ca-42e0-a820-d5761b9458ed.png) | ![after](https://user-images.githubusercontent.com/4588074/155291880-7f1ef576-ed40-4686-a5c0-a737aaa5e145.png) |

## Regression Notes
1. **Potential unintended areas of impact**
    - Layout of the text field component in LTR languages
    - Other places where the text field component is used (I only tested Site Creation → Choose a Domain)
    
2. **What I did to test those areas of impact (or what existing automated tests I relied on)**
✅ Manual testing of the screen in LTR and RTL languages.

3. **What automated tests I added (or what prevented me from doing so)**
I'm not sure if it's applicable to add UI tests for this, I didn't see any in a previous pull request for this input (I.e. : [#13597](https://github.com/wordpress-mobile/WordPress-iOS/pull/13597)), but I'm open to suggestions!

PR submission checklist:
